In [ ]:
import os
U="https://huggingface.co/spaces/TuwaiqAcademy/AISA-ArabicFC-SharedTask-Leaderboard/raw/main"
for f in ["normalize.py","eval_lib.py","data_loader.py"]:
    if not os.path.exists(f): os.system(f"wget -q {U}/{f}")     # download official scorer
from data_loader import parse_predictions, load_gold            # data_loader needs `datasets` (Colab has it)
from eval_lib import evaluate
gold = load_gold("dev")
# for f in ["qwen_final_submission_trackA.jsonl","qwen_final_submission_trackB.jsonl"]:
for f in ["test_submission_trackA_L16_rule_pp.jsonl","test_submission_trackB_L16_rule_pp.jsonl"]:
    preds,_ = parse_predictions(f)
    r = evaluate(preds, gold)
    print(f"{f}: OverallA {r['overall_a']:.4f}  ArgEM {r['argem']:.4f}  OverallB {r['overall_b']:.4f}")

test_submission_trackA_L16_rule_pp.jsonl: OverallA 0.0191  ArgEM 0.0000  OverallB 0.0143
test_submission_trackB_L16_rule_pp.jsonl: OverallA 0.0191  ArgEM 0.0000  OverallB 0.2143


In [ ]:
print(r)

{'fnacc': 0.047706422018348627, 'argem': 0.0, 'thinkrate': 1.0, 'overall_a': 0.019082568807339453, 'overall_b': 0.2143119266055046, 'dialect_breakdown': {'msa': {'fnacc': 0.05263157894736842, 'argem': 0.0, 'n': 323, 'n_positive': 289}, 'gulf': {'fnacc': 0.06666666666666667, 'argem': 0.0, 'n': 75, 'n_positive': 72}, 'levantine': {'fnacc': 0.02666666666666667, 'argem': 0.0, 'n': 75, 'n_positive': 67}, 'egyptian': {'fnacc': 0.015151515151515152, 'argem': 0.0, 'n': 66, 'n_positive': 66}, 'maghrebi': {'fnacc': 0.16666666666666666, 'argem': 0.0, 'n': 6, 'n_positive': 6}}, 'gap_fnacc': 0.1515151515151515, 'gap_argem': 0.0, 'n_total': 545, 'n_positive': 500, 'n_negative': 45, 'n_predictions': 1125, 'missing': 0}


submission_trackA_qwen_lora.jsonl: OverallA 0.9268  ArgEM 0.8780  OverallB 0.7390
submission_trackB_qwen_lora.jsonl: OverallA 0.9268  ArgEM 0.8780  OverallB 0.9225
{'fnacc': 1.0, 'argem': 0.878, 'thinkrate': 0.9174311926605505, 'overall_a': 0.9268, 'overall_b': 0.9224862385321101, 'dialect_breakdown': {'msa': {'fnacc': 1.0, 'argem': 0.8927335640138409, 'n': 323, 'n_positive': 289}, 'gulf': {'fnacc': 1.0, 'argem': 0.9166666666666666, 'n': 75, 'n_positive': 72}, 'levantine': {'fnacc': 1.0, 'argem': 0.8059701492537313, 'n': 75, 'n_positive': 67}, 'egyptian': {'fnacc': 1.0, 'argem': 0.8484848484848485, 'n': 66, 'n_positive': 66}, 'maghrebi': {'fnacc': 1.0, 'argem': 0.8333333333333334, 'n': 6, 'n_positive': 6}}, 'gap_fnacc': 0.0, 'gap_argem': 0.11069651741293529, 'n_total': 545, 'n_positive': 500, 'n_negative': 45, 'n_predictions': 545, 'missing': 0}


submission_trackA_gemm4_e2b.jsonl: OverallA 0.9016  ArgEM 0.8360  OverallB 0.7180
submission_trackB_gemm4_e2b.jsonl: OverallA 0.9016  ArgEM 0.8360  OverallB 0.9085
